### PR TITLE
[DOCS][PYTHON] Update embeddings API docs & Example

### PR DESCRIPTION
**Pull request by AyushExel**
This PR adds an overview of embeddings docs:
- 2 ways to vectorize your data using lancedb - explicit & implicit
- explicit - manually vectorize your data using `wit_embedding` function
- Implicit - automatically vectorize your data as it comes by ingesting your embedding function details as table metadata
- Multi-modal example w/ disappearing embedding function

### CODE REVIEW

**Comment by prrao87** [comment]
Hi @AyushExel I made some edits to other docs in #531 (maintaining your structure as per this PR), so hope that works!

I presume you'd like to keep things clean with edits to the docs so I just thought I'd let you know.

File: docs/src/embeddings/default_embedding_functions.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+registry = EmbeddingFunctionRegistry.get_instance()
+func = registry.get("sentence-transformers").create(device="cpu")
+
+class Words(LanceModel):
+    text: str = func.SourceField()
+    vector: Vector(func.ndims()) = func.VectorField()
+
+table = db.create_table("words", schema=Words)
+table.add(
+    pd.DataFrame(
```
Comment (changhiskhan) [inline docs/src/embeddings/default_embedding_functions.md:27]: let's use list of dict instead for these default examples

File: docs/src/embeddings/embedding_functions.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+table = db.create_table("pets", schema=Pets)
+
+```
+That's it! We have ingested all the information needed to embed source and query inputs. We can now forget about the model and dimension details and start to build or VectorDB
+
+### Step 4 - Ingest lots of data and run vector search!
+Now you can just add the data and it'll be vectorized automatically
+
+```python
+table.add(pd.DataFrame({"image_uri": uris}))
```
Comment (changhiskhan) [inline docs/src/embeddings/embedding_functions.md:43]: i guess use list of dict instead? like `[{"image_uri": u} for u in uris]` or smth without the dataframe

**Review by changhiskhan — COMMENTED** [review]
I forgot how much overlap there is between the two notebooks. Might make sense to either merge them or make them more different?

File: docs/src/embeddings/embedding_functions.md (hunk 2 above)
Comment (AyushExel) [reply docs/src/embeddings/embedding_functions.md]: done

**Comment by AyushExel** [comment]
Yeah, the 2nd notebook is pretty much the same, it wasn't added in the navigation. Removed it. I think we need another non-multimodal notebook example but it can come later as well

**Review by changhiskhan — APPROVED** [review]
Please fix CI before merging

**Pull request merged**